packages/provider/index.js

```diff
@@ -33,33 +34,38 @@ module.exports = {
   },
 
   testConnection: function(options) {
+    const { networks, network } = options;
+    let networkCheckTimeout, networkType;
+    if (networks[network]) {
+      networkCheckTimeout =
+        networks[network].networkCheckTimeout || DEFAULT_NETWORK_CHECK_TIMEOUT;
+      networkType = networks[network].type;
+    } else {
+      networkCheckTimeout = DEFAULT_NETWORK_CHECK_TIMEOUT;
+    }
     const provider = this.getProvider(options);
     const interfaceAdapter = new InterfaceAdapter();
-    const web3 = new Web3Shim({ provider });
+    const web3 = new Web3Shim({ provider, networkType });
     return new Promise((resolve, reject) => {
-      console.log("Testing the provider...");
-      console.log("=======================");
       const noResponseFromNetworkCall = setTimeout(() => {
-        console.log(
-          "> There was a timeout while attempting to connect " +
-            "to the network."
-        );
         const errorMessage =
-          "Failed to connect to the network using the " +
-          "supplied provider.\n       Check to see that your provider is valid.";
+          "There was a timeout while attempting to connect to the network." +
+          "\n       Check to see that your provider is valid.\n       If you " +
+          "have a slow internet connection, try configuring a longer " +
+          "timeout in your Truffle config. Use the " +
+          "networks.<networkName>.networkCheckTimeout property to do this.";
```

Comment:
networks["<networkName>"] is probably better syntax

Reply:
Yes, it is also more consistent with notes in other parts of Truffle